small_language_model_pretraining/nemo/dev/run_llama31.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+#git config --global --add safe.directory /workspace/llama31
+
+# Vars without defaults
+#     Slurm settings
+: "${USER:?USER not set}"
+: "${HOST:?HOST not set}"
+: "${ACCOUNT:?ACCOUNT not set}"
+: "${PARTITION:?PARTITION not set}"
+: "${REMOTE:=0}"
+
+#     Job settings
+: "${JOB_DIR:?JOB_DIR not set}"
+: "${IMAGE:?IMAGE not set}"
+
+#     Dataset settings
+: "${PREPROCESSED_PATH:?PREPROCESSED_PATH not set}"
+: "${TOKENIZER_PATH:?TOKENIZER_PATH not set}"
+
+#     Model settings
+: "${MODEL_CKPT:?MODEL_CKPT not set}"
+: "${USE_CKPT:?USE_CKPT not set}"
+: "${FROM_HF:?FROM_HF not set}"
+: "${CONTINUAL_CKPT:?CONTINUAL_CKPT not set}"
+
+# Vars with defaults
+#     Slurm settings
+: "${TIME:="04:00:00"}"
+: "${NNODES:=1}"
+: "${GPUS_PER_NODE:=8}"
+: "${DEPENDENCIES:=""}"
+
+#     Job settings
+: "${NEMO_DIR:=""}" # Provide customized NeMo path here
+: "${NEMO_RUN_DIR:=""}" # Provide customized NeMo-Run path here
+: "${TMP_NPY_INDEX:=""}" # Provide temporary NNumpy Index saving directory
+: "${MAX_RETRIES:=0}"
+
+#     Model settings
+: "${SIZE:="8b"}"
+: "${GBS:=4}"
+: "${MBS:=1}"
+: "${START_STEPS:=0}"
+
+#     Dataloader settings
+: "${MAX_STEPS:=""}"
```

Comment:
Set MAX_STEPS to 1200000 as it is hardcoded in the benchmark. Remove overwriting MAX_STEPS in individual config files